text/0374-jsii-modular.md

```diff
@@ -0,0 +1,256 @@
+> Minor releases of the `typescript` compiler almost always include syntax
+> breaking changes together with new language features. Since `jsii` now
+> releases in line with `typescript` minor lines, `jsii` no longer adheres to
+> semantic versioning.
+
+The `jsii` release notes for the initial release on each new *major.minor* line
+will include a link to the corresponding TypeScript release notes entry, and a
+description of any `jsii` breaking changes that may have been introduced with
+the release.
+
+When setting up a `jsii` project, we recommend pinning the dependency on the
+`jsii` compiler to the desired minor version line (which corresponds to the
+`typescript` release line), using a `~` SemVer range:
+
+```js
+{
+  // ...
+  "devDependencies": {
+    // ...
+    "jsii": "~4.3.1",
+    // ...
+  },
+  // ...
+}
+```
+````
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+> This section should include answers to questions readers will likely ask about
+> this release. Similar to the "working backwards", this section should be
+> written in a language as if the feature is now released.
+>
+> The template includes a some common questions, feel free to add any questions
+> that might be relevant to this feature or omit questions that you feel are not
+> applicable.
+
+### What are we launching today?
+
+We are announcing a change in versioning strategy for the *jsii* constellation
+of packages, which opens the way for an increased pace of innovation in the
+*jsii* ecosystem:
+
+- Each package of the constellation now has it's own, independent version
+  number, and only gets a new release when a change was introduced;
+- The `jsii` compiler now follows the `typescript`  compiler version number,
+  allowing developers to easily leverage improvements in the *TypeScript*
+  language and compiler performance;
+- The `@jsii/spec` package guarantees all releases of packages in the
+  constellation remain inter-operable going forward.
+
+### Why should I use this feature?
+
+The new versioning strategy allows developers to have greater control over the
+*jsii* tools they use. It allows developers to benefit from improvements to the
+*TypeScript* language (and compiler performance) faster, by allowing them to
+upgrade their project on their own timeline - in the same way they would be able
+to do for a "pure" *TypeScript* project.
+
+Since all parts of the *jsii* constellation are inter-operable with all major
+releases of the *jsii* packages, developers are able to opt into new and
+improved code-generation features of `jsii-pacmak` on their own schedule (i.e:
+when they release a new major version of their library) without requiring all
+their dependencies and dependents use the same version of `jsii-pacmak`.
```

Comment:
Like... exact same version? Do we not check assembly version? How do we even check or validate what version of `pacmak` a dependency was generated with?

Or is `assembly version == jsii version` ? If so, then another "Why Not Just X" that should be addressed is "why not decouple assembly version from jsii version" ?